vcpkg.json

```diff
@@ -0,0 +1,76 @@
+{
+  "name": "keepassxc",
+  "version-string": "2.8.0",
+  "builtin-baseline": "2a6371b01420d8820d158b4707e79931feba27aa",
+  "dependencies": [
+    {
+      "name": "argon2",
+      "version>=": "20190702"
```

Comment:
@phoerious all the versions are defined using a minimum version only. Is there a lock file that vcpkg uses to make sure that the builds are reproducible?

Reply:
There is no lock file AFAIK. But reproducibility could only be achieved if all dependencies are also reproducible themselves, which is probably not the case.

Reply:
> But reproducibility could only be achieved if all dependencies are also reproducible themselves, which is probably not the case.

I agree with that, but pinning the dependencies would nevertheless make it more reproducible that it is at the moment, even if we don't have reproducibility at the binary level. [Apparently dependabot does't support vcpkg](https://github.com/dependabot/dependabot-core/issues/2027), but I think we should consider using an exact version specifier and creating pull requests to bump the dependencies.

Reply:
I'm generally not a fan of package version pinning because it usually results in us forgetting to update the package version before release and being stuck on an unsupported or potentially vulnerable version.

Reply:
> I'm generally not a fan of package version pinning because it usually results in us forgetting to update the package version before release and being stuck on an unsupported or potentially vulnerable version.

That's why I was mentioning dependabot. If it supported vcpkg, we would get notified of new versions through PRs, and we would know if the new version was compatible with our codebase from the result of the CI build. The `develop` builds would not randomly start failing because a new version of a dependency was published between two commits. Without automated PRs for dependency updates however, we'll definitely forget to bump the dependencies.

Also, it looks like the `>=` constraint is already pinning the exact version:
> vcpkg selects the lowest version that matches all constraints, so a less-than constraint is not required.

[source](https://learn.microsoft.com/en-us/vcpkg/users/versioning#version-gte)

Apparently, the `baseline` "provides global minimum version information for your manifest" ([source](https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#builtin-baseline)). So the version will be the oldest of the two minimum versions (the one in the baseline and the one in the manifest). This would mean that as long as we don't change the `baseline` commit or the version specifier, vcpkg would always install the same version.